src/main/webapp/app/text-editor/text-editor.component.ts

```diff
@@ -111,6 +113,22 @@ export class TextEditorComponent implements OnInit {
         );
     }
 
+    ngOnDestroy() {
+        if (this.canDeactivate() && this.textExercise.id) {
+            if (!this.submission) {
+                this.submission = new TextSubmission();
+            }
+            this.submission.submitted = false;
+            this.submission.text = this.answer;
+            this.textSubmissionService[this.submission.id ? 'update' : 'create'](this.submission, this.textExercise.id).subscribe(response => {
```

Comment:
If somebody would e.g. rename `textSubmissionService.update`, this place would not get caught by the refactoring. It's slick but we should use the longer direct method reference instead.

Reply:
Fixed in https://github.com/ls1intum/Artemis/pull/971/commits/d1898947418783f542c86a1ab3adc1cd32649164